examples/go/Dockerfile

```diff
@@ -1,12 +1,16 @@
-FROM golang:1.12 as build-env
+FROM golang:latest as build-env
```

Comment:
sorry I missed this, lets pin this to something: 1.17, it's the style we use in other examples and generally prevents accidental breakages.

Reply:
There are pros and cons to both pinning the version versus defaulting to latest.

Since the other examples pin the version, the change back to a pinned version seems to make sense for coherence.

Updated.